### ISSUE TITLE
Cannot use library with a secured Endpoint

### ISSUE THREAD

**Issue by bytepoets-ssi** [camunda-community-hub/camunda-platform-7-rest-client-java#36]
When using the Spring Boot Starter there is no way to set basic auth credentials for the ApiClient. It is not done in the constructor and the map is immutable after that.

**Comment by berndruecker** [comment]
Good finding and indeed a valuable feature request. Would it be possible for you (@bytepoets-ssi) to provide a pull request for this by chance?

**Comment by janhuddel** [comment]
Hi.

The problem is not this project but the OpenAPI specification in the openapi.json. As far as I can see the artifact is provided as part of the camunda-bpm-platform (https://github.com/camunda/camunda-bpm-platform/tree/master/engine-rest/engine-rest-openapi).

The specification lacks a security scheme in the components: https://swagger.io/docs/specification/authentication/basic-authentication/

If I add this component to the openapi.json and reference it accordingly in the security section, then an ApiClient is generated which supports BasicAuth.

Example (openapi.json):
```
[...]
"security": ["basicAuth"],
  "components": {
    "securitySchemes": {
      "basicAuth": {
      "type": "http", 
      "scheme": "basic"
    }
  },
  "schemas": {
[...]
```

**Comment by berndruecker** [comment]
Happy to release a new versiom once the PR from @janhuddel is through - thank you very much!

**Comment by janhuddel** [comment]
My PR has been accepted in the meantime. The fix will be made available with version 7.19.

See release announcement for 7.19.0-alpha-1: https://camunda.com/blog/2022/11/camunda-automation-platform-7-19-0-alpha1-released/
